lua/keymappings.lua

```diff
@@ -1,83 +1,106 @@
+    -- Terminal window navigation
+    { "<C-h>", "<C-\\><C-N><C-w>h" },
+    { "<C-j>", "<C-\\><C-N><C-w>j" },
+    { "<C-k>", "<C-\\><C-N><C-w>k" },
+    { "<C-l>", "<C-\\><C-N><C-w>l" },
+  },
+  n = { -- Normal mode
+    -- Better window movement
+    { "<C-h>", "<C-w>h", { silent = true } },
+    { "<C-j>", "<C-w>j", { silent = true } },
+    { "<C-k>", "<C-w>k", { silent = true } },
+    { "<C-l>", "<C-w>l", { silent = true } },
 
--- I hate escape
-vim.api.nvim_set_keymap("i", "jk", "<ESC>", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("i", "kj", "<ESC>", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("i", "jj", "<ESC>", { noremap = true, silent = true })
+    -- Resize with arrows
+    { "<C-Up>", ":resize -2<CR>", { silent = true } },
+    { "<C-Down>", ":resize +2<CR>", { silent = true } },
+    { "<C-Left>", ":vertical resize -2<CR>", { silent = true } },
+    { "<C-Right>", ":vertical resize +2<CR>", { silent = true } },
 
--- Tab switch buffer
--- vim.api.nvim_set_keymap("n", "<TAB>", ":bnext<CR>", { noremap = true, silent = true })
--- vim.api.nvim_set_keymap("n", "<S-TAB>", ":bprevious<CR>", { noremap = true, silent = true })
+    -- Tab switch buffer
+    -- { "<TAB>", ":bnext<CR>" },
+    -- { "<S-TAB>", ":bprevious<CR>" },
 
--- Move selected line / block of text in visual mode
-vim.api.nvim_set_keymap("x", "K", ":move '<-2<CR>gv-gv", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("x", "J", ":move '>+1<CR>gv-gv", { noremap = true, silent = true })
+    -- Move current line / block with Alt-j/k a la vscode.
+    { "<A-j>", ":m .+1<CR>==" },
+    { "<A-k>", ":m .-2<CR>==" },
 
--- Move current line / block with Alt-j/k ala vscode.
-vim.api.nvim_set_keymap("n", "<A-j>", ":m .+1<CR>==", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("n", "<A-k>", ":m .-2<CR>==", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("i", "<A-j>", "<Esc>:m .+1<CR>==gi", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("i", "<A-k>", "<Esc>:m .-2<CR>==gi", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("x", "<A-j>", ":m '>+1<CR>gv-gv", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("x", "<A-k>", ":m '<-2<CR>gv-gv", { noremap = true, silent = true })
+    -- QuickFix
+    { "]q", ":cnext<CR>" },
+    { "[q", ":cprev<CR>" },
 
--- QuickFix
-vim.api.nvim_set_keymap("n", "]q", ":cnext<CR>", { noremap = true, silent = true })
-vim.api.nvim_set_keymap("n", "[q", ":cprev<CR>", { noremap = true, silent = true })
+    -- {'<C-TAB>', 'compe#complete()', {noremap = true, silent = true, expr = true}},
+  },
+  t = { -- Terminal mode
+    -- Terminal window navigation
+    { "<C-h>", "<C-\\><C-N><C-w>h" },
+    { "<C-j>", "<C-\\><C-N><C-w>j" },
+    { "<C-k>", "<C-\\><C-N><C-w>k" },
+    { "<C-l>", "<C-\\><C-N><C-w>l" },
+  },
+  v = { -- Visual/Select mode
+    -- Better indenting
+    { "<", "<gv" },
```

Comment:
I think this is needed as well
`{ noremap = true, silent = true }`

Reply:
These are the default options passed to register_mappings

Reply:
:facepalm:  my bad